dashboard/app/models/pd/workshop.rb

```diff
@@ -300,9 +304,23 @@ def self.send_reminder_for_upcoming_in_days(days)
     raise "Failed to send #{days} day workshop reminders: #{errors.join(', ')}" unless errors.empty?
   end
 
+  def self.send_reminder_to_close
+    # Collect errors, but do not stop batch. Rethrow all errors below.
+    errors = []
+    should_have_ended.each do |workshop|
+      begin
+        Pd::WorkshopMailer.organizer_should_close_reminder(workshop).deliver_now
+      rescue => e
+        errors << "organizer should close workshop #{workshop.id} - #{e.message}"
+      end
+    end
+    raise "Failed to send should close workshop reminders: #{errors.join(', ')}" unless errors.empty?
```

Comment:
probably don't need each error to repeat "organizer should close workshop" since that text also appears at the beginning of the message.